File: android/src/main/kotlin/com/sabiantech/sabian_native_common/channels/methods/DeviceChannelHandler.kt
package com.sabiantech.sabian_native_common.channels.methods

class DeviceChannelHandler : AbstractMethodChannelHandler() {
    override fun register() {
        register("getPlatformVersion", object : IMethodChannelHandler {
            override fun execute(payload: MethodChannelPayload) {
                payload.result.success("Android ${android.os.Build.VERSION.RELEASE}")
            }
        })
    }
}

File: sabian_native_common_ios/ios/Classes/channels/methods/DeviceChannelHandler.swift
//
//  DeviceChannelHandler.swift
//  sabian_native_common
//
//  Created by bryosabian on 06/12/2023.
//

import Foundation

class DeviceChannelHandler : AbstractMethodChannelHandler {
    override func register() {
        register("getPlatformVersion", PlatformVersionMethod())
    }
}

private class PlatformVersionMethod : IMethodChannelHandler {
    func execute(payload: MethodChannelPayload) {
        payload.result("iOS " + UIDevice.current.systemVersion)
    }
}
